src/plugins/vis_augmenter/public/vega/helpers.ts

```diff
@@ -340,3 +344,98 @@ export const addPointInTimeEventsLayersToSpec = (
 
   return newSpec;
 };
+
+// This is the total y-axis padding such that if this is added to the "padding" value of the view, if there is no axis,
+// it will align values on the x-axis
+export const calculateYAxisPadding = (config: YAxisConfig): number => {
+  // TODO: figure out where this value is coming from
+  const defaultPadding = 3;
+  return (
+    get(config, 'minExtent', 0) +
+    get(config, 'offset', 0) +
+    get(config, 'translate', 0) +
+    get(config, 'domainWidth', 0) +
+    get(config, 'labelPadding', 0) +
+    get(config, 'titlePadding', 0) +
+    get(config, 'tickOffset', 0) +
+    get(config, 'tickSize', 0) +
+    defaultPadding
+  );
+};
+
+// Parse the vis augmenter config to apply different visual changes to the event chart spec.
+// This includes potentially removing the original vis data, hiding axes, moving the legend, etc.
+// Primarily used within the view events flyout to render the charts in different ways, and to
+// ensure the stacked event charts are aligned with the base vis chart.
+export const augmentEventChartSpec = (
+  config: VisAugmenterEmbeddableConfig,
+  origSpec: object
+): {} => {
+  const inFlyout = get(config, 'inFlyout', false) as boolean;
+  const flyoutContext = get(config, 'flyoutContext', VisFlyoutContext.BASE_VIS);
+
+  const newVconcat = [] as Array<{}>;
+  // @ts-ignore
+  const newConfig = origSpec?.config;
+  const visChart = get(origSpec, 'vconcat[0]', {});
+  const eventChart = get(origSpec, 'vconcat[1]', {});
+
+  if (inFlyout) {
+    switch (flyoutContext) {
+      case VisFlyoutContext.BASE_VIS:
+        newConfig.legend = {
+          ...newConfig.legend,
+          orient: 'top',
+          // need to set offset to 0 so we don't cut off the chart canvas within the embeddable
+          offset: 0,
+        };
+        break;
+
+      case VisFlyoutContext.EVENT_VIS:
+        eventChart.encoding.x.axis = {
+          domain: true,
+          grid: false,
+          ticks: false,
+          labels: false,
+          title: null,
+        };
+        eventChart.mark.fillOpacity = 0;
+        break;
+
+      case VisFlyoutContext.TIMELINE_VIS:
+        eventChart.transform = [
+          {
+            filter: 'false',
+          },
+        ];
+        break;
+    }
+
+    // if coming from view events page, need to standardize the y axis padding values so we can
+    // align all of the charts correctly
+    newConfig.axisY = {
+      // We need minExtent and maxExtent to be the same. We cannot calculate these on-the-fly
+      // so we need to force a static value.
```

Comment:
Can we add a comment that we choose these specific static values to make the alignment right?

Reply:
done